validator/validators/performance/performance_validator.go

```diff
@@ -131,6 +147,42 @@ func (s *PerformanceValidator) CalculateMetricStatsAndPackMetrics(metrics []type
 	return packIntoPerformanceInformation(uniqueID, receiver, dataType, fmt.Sprint(agentCollectionPeriod), commitHash, commitDate, map[string]interface{}{dataRate: performanceMetricResults}), nil
 }
 
+func (s *PerformanceValidator) CalculateWindowsMetricStatsAndPackMetrics(statistic []*cloudwatch.GetMetricStatisticsOutput) (PerformanceInformation, error) {
+	var (
+		receiver               = s.vConfig.GetPluginsConfig()[0] //Assuming one plugin at a time
+		commitHash, commitDate = s.vConfig.GetCommitInformation()
+		dataType               = s.vConfig.GetDataType()
+		dataRate               = fmt.Sprint(s.vConfig.GetDataRate())
+		uniqueID               = s.vConfig.GetUniqueID()
+		agentCollectionPeriod  = s.vConfig.GetAgentCollectionPeriod().Seconds()
+	)
+	performanceMetricResults := make(map[string]Stats)
+
+	for _, metric := range statistic {
+		metricLabel := strings.Split(*metric.Label, " ")
+		metricName := metricLabel[len(metricLabel)-1]
+		metricValues := metric.Datapoints
+		//Convert every bytes to MB
+		if slices.Contains(metricsConvertToMB, metricName) {
+			for i, val := range metricValues {
+				*metricValues[i].Average = *val.Average / (1024 * 1024)
+			}
+		}
+		log.Printf("Start calculate metric statictics for metric %s \n", metricName)
+		if !isAllStatisticsGreaterThanOrEqualToZero(metricValues) {
+			return nil, fmt.Errorf("\n values are not all greater than or equal to zero for metric %s with values: %v", metricName, metricValues)
+		}
+		// GetMetricStatistics provides these statistics, however this will require maintaining multiple data arrays
```

Comment:
Does metric rename not work for these?

Reply:
Double checked offline that metric decoration apparently is not supported for procstat so this comment can be disregarded :(

Reply:
Not sure why this is the case. May be worth to investigate for the future since I can see this being a potential issue.